File: src/Java/testapp/src/main/java/com/myservicebus/testapp/SubmitOrderFaultConsumer.java
package com.myservicebus.testapp;

import java.util.concurrent.CompletableFuture;

import com.google.inject.Inject;
import com.myservicebus.ConsumeContext;
import com.myservicebus.Consumer;
import com.myservicebus.Fault;
import com.myservicebus.logging.Logger;
import com.myservicebus.logging.LoggerFactory;

class SubmitOrderFaultConsumer implements Consumer<Fault<SubmitOrder>> {
    private final Logger logger;

    @Inject
    public SubmitOrderFaultConsumer(LoggerFactory loggerFactory) {
        this.logger = loggerFactory.create(SubmitOrderFaultConsumer.class);
    }

    @Override
    public CompletableFuture<Void> consume(ConsumeContext<Fault<SubmitOrder>> context) throws Exception {
        var fault = context.getMessage();
        var msg = fault.getMessage();
        String error = fault.getExceptions().isEmpty() ? "" : fault.getExceptions().get(0).getMessage();
        logger.warn("⚠️ SubmitOrder fault for {}: {}", msg.getOrderId(), error);
        return CompletableFuture.completedFuture(null);
    }
}

File: src/TestApp/SubmitOrderFaultConsumer.cs
using Microsoft.Extensions.Logging;
using MyServiceBus;

namespace TestApp;

class SubmitOrderFaultConsumer : IConsumer<Fault<SubmitOrder>>
{
    private readonly ILogger<SubmitOrderFaultConsumer> _logger;

    public SubmitOrderFaultConsumer(ILogger<SubmitOrderFaultConsumer> logger)
    {
        _logger = logger;
    }

    public Task Consume(ConsumeContext<Fault<SubmitOrder>> context)
    {
        var fault = context.Message;
        var message = fault.Message;
        var error = fault.Exceptions.Count > 0 ? fault.Exceptions[0].Message : "Unknown error";

        _logger.LogWarning("⚠️ SubmitOrder fault for {OrderId}: {Error}", message.OrderId, error);
        return Task.CompletedTask;
    }
}
